In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np


In [2]:
lats = np.random.uniform(-90.000,90.000,1500)
lngs = np.random.uniform(-180.000,180.000,1500)
lat_lngs = zip(lats, lngs)
lat_lngs

In [3]:
# Add the latitudes and longitudes to a list
coordinates = list(lat_lngs)

In [4]:
# Now, they are a list of tuples
# Now, we can iterate through it to find the closest city to that coordinate
coordinates

[(-80.64692981513878, 35.59034663203417),
 (59.50644066066138, 90.35650223237548),
 (-89.56032887675305, -176.67807739080678),
 (-35.9201776455181, 38.41662456989488),
 (51.61016857632481, -51.21310967922906),
 (-0.9820415534075551, 138.21153280651214),
 (-9.771636814401077, -163.0880258969652),
 (82.70512131467979, 81.44621726841922),
 (47.017022270460075, 103.36158852799872),
 (-39.729401653028795, -88.66996376059312),
 (-22.25119038101252, -137.05841423514505),
 (40.01955293341885, -79.60511935220121),
 (-9.49301418251828, 141.96284772653024),
 (-20.69252423302423, 154.61346000807703),
 (61.27404763083743, 17.553025972291834),
 (28.71748056101596, 149.57015944808165),
 (-49.82273781609723, -129.87476183098897),
 (20.579432479808915, -22.69335082593676),
 (34.11808480668938, -60.97128240944474),
 (-8.142018595476017, 166.92465907814687),
 (64.54926309897536, 145.6259736233785),
 (10.327450550946153, -18.33658287143885),
 (-62.85390583245736, -161.230145334018),
 (16.6646710181288, 61

In [6]:
from citipy import citipy 

In [7]:
# Create a list for holding the cities
cities = []
# Identify the nearest city for each latitude and longitude combination
for coordinate in coordinates:
    city = citipy.nearest_city(coordinate[0], coordinate[1]).city_name
    
    # If the city is unique, then we will add it to the cities list
    if city not in cities:
        cities.append(city)
        
len(cities)
    

620

This is basically saying that from the 1500 GCS locations we had, only 620 came from it as there were some repeated that were close to the same random location. Remember that we were trying to come up with more than 500 hundred cities and we did this by setting 1500 GSC locations knowing that almost 70% of the earth is covered with water. The other 30% were our chances of getting a specific city. 